src/main/java/com/yugabyte/sample/apps/CassandraEventData.java

```diff
@@ -0,0 +1,309 @@
+		// Set the number of keys to read and write.
+		appConfig.numKeysToRead = -1;
+		appConfig.numKeysToWrite = -1;
+		// The size of the value.
+		appConfig.valueSize = 3000;
+		// Set the TTL for the raw table.
+		appConfig.tableTTLSeconds = 24 * 60 * 60;
+		// Default write batch size.
+		appConfig.cassandraBatchSize = 25;
+		// Default read batch size.
+		appConfig.cassandraReadBatchSize = 25;
+		// Default time delta from current time to read in a batch.
+		appConfig.readBackDeltaTimeFromNow = 180;
+	}
+
+	static Random random = new Random();
+	// The default table name that has the raw metric data.
+	private final String DEFAULT_TABLE_NAME = "batch_event_data_ts_metrics_raw";
+	// The structure to hold info per metric.
+	static List<DataSource> dataSources = new CopyOnWriteArrayList<DataSource>();
+	// The minimum number of metrics to simulate.
+	private static int min_metrics_count = 50;
+	// The maximum number of metrics to simulate.
+	private static int max_metrics_count = 100;
+	// The shared prepared select statement for fetching the data.
+	private static volatile PreparedStatement preparedSelect;
+	// The shared prepared statement for inserting into the table.
+	private static volatile PreparedStatement preparedInsert;
+	// Lock for initializing prepared statement objects.
+	private static final Object prepareInitLock = new Object();
+
+	@Override
+	public void initialize(CmdLineOpts configuration) {
+		synchronized (dataSources) {
+			// If the datasources have already been created, we have already initialized the
+			// static
+			// variables, so nothing to do.
+			if (!dataSources.isEmpty()) {
+				return;
+			}
+
+			// Read the various params from the command line.
+			CommandLine commandLine = configuration.getCommandLine();
+			if (commandLine.hasOption("min_metrics_count")) {
+				min_metrics_count = Integer.parseInt(commandLine.getOptionValue("min_metrics_count"));
+			}
+			if (commandLine.hasOption("max_metrics_count")) {
+				max_metrics_count = Integer.parseInt(commandLine.getOptionValue("max_metrics_count"));
+			}
+
+			// Generate the number of metrics this data source would emit.
+			int num_metrics = min_metrics_count
+					+ (max_metrics_count > min_metrics_count ? random.nextInt(max_metrics_count - min_metrics_count)
+							: 0);
+
+			// Create all the metric data sources.
+			for (int i = 0; i < num_metrics; i++) {
+				DataSource dataSource = new DataSource(i);
+				dataSources.add(dataSource);
+			}
+		}
+	}
+
+	public String getTableName() {
+		return appConfig.tableName != null ? appConfig.tableName : DEFAULT_TABLE_NAME;
+	}
+
+	@Override
+	public void dropTable() {
+		dropCassandraTable(getTableName());
+	}
+
+	@Override
+	protected List<String> getCreateTableStatements() {
+		
+		return Arrays.asList(
+		        String.format("CREATE TABLE IF NOT EXISTS %s %s %s %s;", getTableName()," ( metric_id varchar , ts bigint , event_type varchar , value blob , primary key (metric_id, ts, event_type))"
```

Comment:
metric_id -> device_id

Also add DESC for timestamp.

Reply:
Done, all the references are now to a Device